dap-mode.el

```diff
@@ -1240,7 +1241,14 @@ DEBUG-SESSION is the active debug session."
              #'equal))
 
 (defun dap--set-exception-breakpoints (debug-session callback)
-  (-let [(&dap-session 'initialize-result 'launch-args (&plist :type)) debug-session]
+  ;; XXX: for whatever reason, make byte-compile emits warnings when
```

Comment:
most likely the definition of the destructoring function should be in eval-when-compile block